tool/clean/clean_host/clean_host.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build clean
+// +build clean
+
+package main
+
+import (
+	"context"
+	"github.com/aws/amazon-cloudwatch-agent/tool/clean"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go/aws"
+	"log"
+	"time"
+)
+
+// Clean integration hosts if they have been open longer than 1 day
+const tagName = "tag:Name"
+var tagValue = []string{
+	"buildLinuxPackage",
+	"buildPKG",
+	"buildMSI",
+	"MSIUpgrade_*",
+	"IntegrationTestBase",
+	"CWADockerImageBuilderX86",
+	"CWADockerImageBuilderARM64",
+}
+func main() {
+	err := cleanHost()
+	if err != nil {
+		log.Fatalf("errors cleaning %v", err)
+	}
+}
+
+func cleanHost() error {
+	log.Print("Begin to clean EC2 Host")
+
+	expirationDateHost := time.Now().UTC().Add(clean.KeepDurationOneDay)
+	cxt := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(cxt)
+	if err != nil {
+		return err
+	}
+	ec2client := ec2.NewFromConfig(defaultConfig)
+
+	hosts := getHost(cxt, ec2client)
+
+	hostIds := make([]string, 0)
+	for _, host := range hosts {
+		log.Printf("host id %v experation date %v host creation date raw %v host state %v",
+			*host.InstanceId, expirationDateHost, *host.LaunchTime, host.State)
+		if expirationDateHost.After(*host.LaunchTime) {
+			log.Printf("Try to delete host %s tags %v launch-date %s", *host.InstanceId, host.Tags, *host.LaunchTime)
+			hostIds = append(hostIds, *host.InstanceId)
+		}
+	}
+
+	if len(hostIds) == 0 {
+		log.Printf("No hosts to terminate")
+		return nil
+	}
+
+	log.Printf("Hosts to temrinate %v", hostIds)
+	terminateHost := ec2.TerminateInstancesInput{InstanceIds: hostIds}
+	_, err = ec2client.TerminateInstances(cxt, &terminateHost)
+	return err
+}
+
+func getHost(cxt context.Context, ec2client *ec2.Client) []types.Instance {
+	// Get list of host
+	nameFilter := types.Filter{Name: aws.String(tagName), Values: tagValue}
+
+	nextToken := aws.String("")
+	hosts := make([]types.Instance, 0)
+	hostInput := ec2.DescribeInstancesInput{Filters: []types.Filter{nameFilter, {Name: aws.String("instance-state-name"), Values: []string{"running"}}}}
+	for {
+		describeHostOutput, _ := ec2client.DescribeInstances(cxt, &hostInput)
+		for _, reservation := range describeHostOutput.Reservations {
+			for _, host := range reservation.Instances {
```

Comment:
It would be better to confirm the instance[ state code](https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html) for avoid adding unnecessary instances host.
```
instance-state-code - The state of the instance, as a 16-bit unsigned integer. The high byte is used for internal purposes and should be ignored. The low byte is set based on the state represented. The valid values are: 0 (pending), 16 (running), 32 (shutting-down), 48 (terminated), 64 (stopping), and 80 (stopped).
```

Reply:
Not needed since we filter by running state.